cmd/internal/flowmode/cmd_run.go

```diff
@@ -346,15 +351,73 @@ func getEnabledComponentsFunc(f *flow.Flow) func() map[string]interface{} {
 	}
 }
 
-func loadFlowFile(filename string) (*flow.File, error) {
-	bb, err := os.ReadFile(filename)
-	if err != nil {
-		return nil, err
+func loadFlowFile(pathName string) (*flow.File, map[string][]byte, error) {
+	var (
+		files        []*flow.File
+		fileContents = map[string][]byte{}
+	)
+
+	filepath.WalkDir(pathName, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		} else if d.IsDir() && path != pathName {
+			return filepath.SkipDir
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		if strings.HasSuffix(path, ".river") || path == pathName {
+			bb, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+			fileContents[path] = bb
+
+			// TODO(rfratto): this breaks when there's multiple files
+			instrumentation.InstrumentConfig(bb)
```

Comment:
Note that multiple files breaks InstrumentConfig, since it's only meant to track a single config at a time, and we'd need another solution for folders.